public/app/core/components/NavBar/NavBarItem.tsx

```diff
@@ -50,7 +50,9 @@ const NavBarItem = ({
     }
 
     if (!target && url.startsWith('/')) {
-      locationService.push(url);
+      // when config.appSubUrl is configured the locationsService uses appSubUrl on all urls, so we need to strip it before push
+      const path = config.appSubUrl ? url.replace(config.appSubUrl, '') : url;
+      locationService.push(path);
     } else {
```

Comment:
I am confused by this NavBarItem onNavigate, this is just a normal link, why all this code to navigate to a link?  The code above we already handle in Link component and for normal anchors we have a global intercept as well 

@ashharrison90  ?

Reply:
@torkelo this is an unfortunate side effect of how react-aria works. They have a very opinionated API where menu items can't contain actionable items so the solution they offer is to use `onAction` here https://github.com/grafana/grafana/blob/7a99b067ef80cd7394bc4ad4326dc4d58a0d8014/public/app/core/components/NavBar/NavBarItemMenuItem.tsx#L42

We did a hacky solution to get around the react-araia limitation before, but we decided to go with their solution with `onAction` in the end:
https://github.com/grafana/grafana/pull/42113/commits/c2fcbfc5ff15ea1ac0e2300ac2360fdfaf55e36f#diff-71fa5f7d566f7c8edcaab336af143a8adca3fa974120cc1978ea48bc2d269d42R61

See the `NOTE: menu items cannot contain interactive content (e.g. buttons, checkboxes, etc.).` here:
https://react-spectrum.adobe.com/react-aria/useMenu.html#complex-menu-items

Reply:
If the solution offered by this PR is worse than the hack in https://github.com/grafana/grafana/pull/42113/commits/c2fcbfc5ff15ea1ac0e2300ac2360fdfaf55e36f#diff-71fa5f7d566f7c8edcaab336af143a8adca3fa974120cc1978ea48bc2d269d42R61 we can revert to that solution.

Reply:
Ok, can't say I understand any of that, looks extremely complicated for handling a focus of a link :exploding_head:

Reply:
It handles much more than focus for a link, it handles all the aria*, keyboard navigation and press events for mobile views. But it's indeed very complex and opinionated.

Reply:
Fixed with suggestion